python/smqtk/algorithms/nn_index/flann.py

```diff
@@ -310,10 +310,9 @@ def _build_index(self, descriptors):
                 fd, fp = tempfile.mkstemp()
                 try:
                     self._flann.save_index(fp)
-                    self._index_elem.set_bytes(os.read(fd,
-                                                       os.path.getsize(fp)))
+                    with os.fdopen(fd, 'rb') as f:
```

Comment:
My thought was that since the file descriptor was already available we ought to use it. It also has the effect of automatically closing the file descriptor on exiting the `with` block so that it doesn't need to be done explicitly in the `finally` block.

Reply:
Sounds good to me.  Maybe adding a rendition of your response as a comment above the use of ``os.fdopen`` will help us understand the decision made here in the future.

Reply:
Done. I could also see extracting a utility function for the "save to a temp file and read its bytes" dance, though it might not be worth it.

With something like this:
```python
def save_to_file_and_read(save):
    """Call save with the path of a temporary file and then return the
    resulting bytes.  This allows wrapping a function that can only
    save to a file.

    """
    fd, fp = tempfile.mkstemp()
    try:
        save(fp)
        with os.fdopen(fd, 'rb') as f:
            return f.read()
    finally:
        os.remove(fp)
```

you would then do:
```python
index_bytes = save_to_file_and_read(lambda fp: faiss.write_index(to_write, fp))
self._index_elem.set_bytes(index_bytes)
```

Reply:
I don't think we need the utility function as of yet since its purpose is still pretty specific.